File: Domain/TransApp.Domain/Shipment/ShipmentModel.cs
﻿using System;
using System.Collections.Generic;
using TransApp.Domain.Addresses;
using TransApp.Domain.Common.Entity;

namespace TransApp.Domain.Shipment
{
    public class ShipmentModel
    {
        public int Id { get; set; }

        public string Reference { get; set; }

        public string PoNumber { get; set; }

        public DateTime? PickUpDate { get; set; }

        public DateTime? DeliveryDate { get; set; }

        public int? CustomerId { get; set; }

        public int? SenderAddressId { get; set; }

        public string SenderContactPerson { get; set; }

        public string SenderPhone { get; set; }

        public string SenderRemark { get; set; }



        public int? ReceiverAddressId { get; set; }

        public string ReceiverContactPerson { get; set; }

        public string ReceiverPhone { get; set; }

        public string ReceiverRemark { get; set; }



        public decimal? TotalPrice { get; set; }

        public decimal? TotalVolume { get; set; }

        public decimal? TotalQuatity { get; set; }

        public decimal? TotalWeight { get; set; }

        public string ShipmentStatus { get; set; }

        public int? TransporterId { get; set; }

        public int? UserIdCreated { get; set; }

        public DateTime? DateCreated { get; set; }

        public int? UserIdModified { get; set; }

        public string UserModified { get; set; }

        public string UserCreated { get; set; }
        public string TransporterName { get; set; }
        public string AddressFrom { get; set; }
        public string AddressTo { get; set; }
        public DateTime? DateModified { get; set; }
        public decimal? InvoiceAmount { get; set; }
        public string InvoiceComment { get; set; }
        public DateTime? InvoiceDate { get; set; }
        public string PickUpComment { get; set; }
        public string DeliveryComment { get; set; }
        public bool? DeliveryPod { get; set; }

        public List<ShipmentDetailModel> ShipmentDetails { get; set; }

        public List<FacilityEntityModel> ReceiverFacilities { get; set; }
        public List<RequirementEntityModel> ReceiverRequirements { get; set; }
        public List<TruckEntityModel> ReceiverTrucks { get; set; }

        public List<FacilityEntityModel> SenderFacilities { get; set; }
        public List<RequirementEntityModel> SenderRequirements { get; set; }
        public List<TruckEntityModel> SenderTrucks { get; set; }

        public List<ShipmentTransporterModel> Transporters { get; set; }
        public List<AvailabilityEntityModel> ReceiverAvailabilities { get; set; }

        public List<AvailabilityEntityModel> SenderAvailabilities { get; set; }
        public int OfferCount { get; set; }
    }
}

File: Site/Presentation/TransApp.Site.FrontEnd/src/app/shipment/models/shipment-model.ts
import { AvailabilityEntityModel } from 'app/shared/common/models/entity/availability-entity-model';
import { ShipmentTransporterModel } from 'app/shipment/models/shipment-transporter-model';
import { ShipmentDetailModel } from 'app/shipment/models/shipment-detail-model';
import { FacilityEntityModel } from 'app/shared/common/models/entity/facility-entity-model';
import { TruckEntityModel } from 'app/shared/common/models/entity/truck-entity-model';
import { RequirementEntityModel } from 'app/shared/common/models/entity/requirement-entity-model';

export class ShipmentModel {
  public id: number;
  public reference: string;
  public poNumber: string;
  public pickUpDate: Date;
  public deliveryDate: Date;
  public customerId: number;
  // sender region
  public senderAddressId: number;
  public senderAddressInfo: string;
  public addressFrom: string;
  public senderAddressName: string;
  public senderContactPerson: string;
  public senderPhone: string;
  public senderRemark: string;
  public senderAvailabilities: AvailabilityEntityModel[];
  public senderFacilities: Array<FacilityEntityModel>;
  public senderRequirements: Array<RequirementEntityModel>;
  public senderTrucks: Array<TruckEntityModel>;
  // receiver region
  public receiverAddressId: number;
  public receiverAddressName: string;
  public receiverAddressInfo: string;
  public addressTo: string;
  public receiverContactPerson: string;
  public receiverPhone: string;
  public receiverRemark: string;
  public receiverAvailabilities: AvailabilityEntityModel[];
  public receiverFacilities: Array<FacilityEntityModel>;
  public receiverRequirements: Array<RequirementEntityModel>;
  public receiverTrucks: Array<TruckEntityModel>;

  // open market
  public offerCount: number;

  public totalPrice: number;
  public totalVolume: number;
  public totalQuatity: number;
  public totalWeight: number;
  public shipmentStatus: string;
  public transporterId: number;
  public shipmentDetails: Array<ShipmentDetailModel>;
  public shipmentTransporters: Array<ShipmentTransporterModel>;

  public userIdCreated: number;
  public dateCreated: Date;
  public userIdModified: number;
  public userModified: string;
  public userCreated: string;
  public dateModified: Date;
}
